torch/_dynamo/variables/tensor.py

```diff
@@ -723,17 +749,11 @@ def call_method(
         from . import GetAttrVariable
 
         options = VariableTracker.propagate(self, args, kwargs.values())
-        # insert unwrapped version of self as the first argument
-        # TODO: This is wrong!  When you call the internal __torch_function__,
-        # you still get the wrapped version of self, and if you call functions
-        # inside __torch_function__, they should come back here.  If we unwrap
-        # the tensor immediately, that will not happen.
-        # See https://github.com/pytorch/torchdynamo/issues/1951
         args = list(args)
-        args.insert(0, self.tensor_variable)
+        args.insert(0, self)
         func_var = GetAttrVariable(self.tensor_variable, name)
```

Comment:
This is pre-existing code, but what exactly is this doing?

Reply:
This was shuffling args around because the override code expected the class to be unwrapped already and be the first argument to `__torch_function__`. It was all arbitrary and needed to be cleaned up

Reply:
I'm talking specifically about the GetAttrVariable construction here

Reply:
Ah, this is getting a variable for the method getting called from the internal tensor var. I could call var_getattr directly, but this is equivalent.